src/pages/Swap/index.tsx

```diff
@@ -768,8 +862,14 @@ export default function Swap() {
                       }
                     }}
                     id="swap-button"
-                    disabled={!isValid || routeIsSyncing || routeIsLoading || priceImpactTooHigh || !!swapCallbackError}
-                    error={isValid && priceImpactSeverity > 2 && !swapCallbackError}
+                    disabled={
+                      !isValid ||
+                      routeIsSyncing ||
+                      routeIsLoading ||
+                      priceImpactTooHigh ||
+                      Boolean(!permit2Enabled && swapCallbackError)
```

Comment:
I think this is complicated enough to warrant moving it up out of the template

Reply:
I think moving it out to a new variable will be equally as confusing.